src/caret_analyze/value_objects/node.py

```diff
@@ -231,15 +231,20 @@ def get_service(
 
     def get_publisher(
         self,
-        publish_topic_name: str
+        publish_topic_name: str,
+        construction_order: Optional[int]
     ) -> PublisherStructValue:
         try:
-            return Util.find_one(
-                lambda x: x.topic_name == publish_topic_name,
-                self._publishers)
+            def is_target_publisher(publisher: PublisherStructValue):
+                return publisher.topic_name == publish_topic_name and \
+                    publisher.construction_order == construction_order
```

Comment:
- I think it is better to treat as "matched" when `construction_order` argument is None.
- Is there any reason for modifying only get_publisher? I think it is better to modify get_subscription, get_service and get_timer.

Reply:
Not only get_publisher, but all of get_subscription, get_service and get_timer failed with the current test code and script.
So I don't remember why only get_publisher has construction_order modification.
It may have been entered by searching with get_publisher.
Maybe get_XXX of this class is not used.
Or maybe there just wasn't enough data to pass.
I will postpone the investigation of this code for a moment

Reply:
I corresponded.

Reply:
I confirmed that `construction_order` is added to another functions.

Can you also consider about my first comment?
> I think it is better to treat as "matched" when construction_order argument is None.

Reply:
Matched if the construction_order argument was None .
Also, I created a new test code for this issue for each file.